ci/run_locally.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+set -eEuo pipefail
+
+root_dir="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")/.." &>/dev/null && pwd)"
+common_file_location="${root_dir}/ci/common.sh"
+workflows_dir="${root_dir}/.github/workflows"
+PRE_PUSH_HOOK="${PRE_PUSH_HOOK:-false}"
+
+
+####
+# Checking all the requirement(s)
+####
+if ! [ -f "${common_file_location}" ]; then
+  echo -e "\n\033[1;31mERROR: ${common_file_location} file is missing !!!  Exiting ...\033[0m\n"
+  exit 1
+else
+  # shellcheck disable=SC1090
+  source "${common_file_location}"
+fi
+check_requirements
+
+# Starting from fresh act environment
+act_dangling_containers="$(docker ps -a -q -f "name=act")" || fn_die "ERROR: Failed to execute the 'docker ps' command. Please ensure the 'docker' service is running. Exiting ..."
+if [ -n "${act_dangling_containers}" ]; then
+  log_info "\n=== Cleaning up dangling 'act' containers ==="
+  docker rm -f "${act_dangling_containers}" || fn_die "ERROR: Failed to execute the 'docker rm -f' command. Please ensure the 'docker' service is running. Exiting ..."
+fi
+
+
+####
+# Running workflow(s)
+####
+workflows_orchestrator="CI-build-test CI-coverage CI-lint-format CI-e2e-test"
+workflows_extra="CI-rustdoc"
+if [ "${PRE_PUSH_HOOK}" == 'false' ];then
+  workflows="${workflows_orchestrator} ${workflows_extra}"
+  while true; do
+    # choose one of the available workflows
+    log_warn "\nPlease select a workflow to run:"
```

Comment:
If I am not wrong it seems to me that in order to make sure a commit is fine one should run the script at least 4 times (CI-build-test, CI-coverage, CI-lint-format, CI-e2e-test; CI-rustdoc can be safely skipped); this can be quite tedious, isn't it possible to queue together all those workflow and have a single long run?

Reply:
@JackPiri yes. In order to run the whole orchestrator workflow it needs to run individual workflows 1 by 1. Running them all in parallel on local env is not stable, but if you already have all the cache artifacts build it will not take long at all.